java-manual/modules/ROOT/pages/transactions.adoc

```diff
@@ -0,0 +1,487 @@
+
+Explicit transactions are most useful for applications that need to distribute Cypher execution across multiple functions for the same transaction, or for applications that need to run multiple queries within a single transaction but without the automatic retries provided by managed transactions.
+
+.An explicit transaction example involving an external API
+[source, java]
+----
+package demo;
+
+import java.util.Map;
+import java.util.List;
+import java.util.Arrays;
+
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.QueryConfig;
+import org.neo4j.driver.Record;
+import org.neo4j.driver.SessionConfig;
+import org.neo4j.driver.Transaction;
+
+public class App {
+
+    private static final String URI = "neo4j://localhost";
+    private static final String USER = "neo4j";
+    private static final String PASSWORD = "verysecret";
+
+    public static void main(String... args) {
+        try (var driver = GraphDatabase.driver(URI, AuthTokens.basic(USER, PASSWORD))) {
+            String customerId = createCustomer(driver);
+            int otherBankId = 42;
+            transferToOtherBank(driver, customerId, otherBankId, 999);
+        }
+    }
+
+    static String createCustomer(Driver driver) {
+        var result = driver.executableQuery("""
+            MERGE (c:Customer {id: randomUUID(), balance: 1000})
+            RETURN c.id AS id
+            """)
+            .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+            .execute();
+        return result.records().get(0).get("id").asString();
+    }
+
+    static void transferToOtherBank(Driver driver, String customerId, int otherBankId, float amount) {
+        try (var session = driver.session(SessionConfig.builder().withDatabase("neo4j").build())) {
+            try (var tx = session.beginTransaction()) {
+                if (! customerBalanceCheck(tx, customerId, amount)) {
+                    System.out.printf("Customer %s doesn't have enough funds.\n", customerId);
+                    return;  // give up
+                }
+
+                otherBankTransferApi(customerId, otherBankId, amount);
+                // Now the money has been transferred => can't rollback anymore
+                // (cannot rollback external services interactions)
+
+                try {
+                    decreaseCustomerBalance(tx, customerId, amount);
+                    tx.commit();
+                    System.out.printf("Transferred %f to %s.\n", amount, customerId);
+                } catch (Exception e) {
+                    requestInspection(customerId, otherBankId, amount, e);
+                    throw new RuntimeException(e.getMessage());
+                } finally {
+                    tx.close();  // rolls back if not yet committed
```

Comment:
Superflous, it is already wrapper in try with resources. The `finally` block should go.